File: mvp-cli/modules/formatting.py
"""Module for storing common terminal styling strings."""
from colorama import Fore, Style

HEAD_STYLE = f"{Style.RESET_ALL}{Style.BRIGHT}{Fore.CYAN}"
BOLD_STYLE = f"{Style.RESET_ALL}{Style.BRIGHT}{Fore.WHITE}"
NORMAL_STYLE = f"{Style.RESET_ALL}{Style.NORMAL}{Fore.WHITE}"
DIM_STYLE = f"{Style.RESET_ALL}{Style.DIM}{Fore.WHITE}"
DEBUG_STYLE = f"{Style.RESET_ALL}{Style.BRIGHT}{Fore.RED}"

BULLET = f"{Style.RESET_ALL}{Style.BRIGHT}{Fore.CYAN}•{Style.RESET_ALL}"
CARET = f"{Style.RESET_ALL}{Style.BRIGHT}{Fore.RED}›{Style.RESET_ALL}"

File: core-lib/src/formatting.cpp
#include "formatting.hpp"

#include <termcolor/termcolor.hpp>

namespace shabam {
namespace formatting {

std::ostream &normal_style(std::ostream &os) {
  os << termcolor::reset << termcolor::white;
  return os;
};

std::ostream &bold_style(std::ostream &os) {
  os << termcolor::reset << termcolor::bold << termcolor::white;
  return os;
};

std::ostream &dim_style(std::ostream &os) {
  os << termcolor::reset << termcolor::dark << termcolor::white;
  return os;
};

std::ostream &warning_style(std::ostream &os) {
  os << termcolor::reset << termcolor::dark << termcolor::yellow;
  return os;
};

std::ostream &bullet(std::ostream &os) {
  os << termcolor::reset << termcolor::bold << termcolor::cyan << "•"
     << termcolor::reset;
  return os;
};

std::ostream &reset_style(std::ostream &os) {
  os << termcolor::reset;
  return os;
}

std::ostream &caret(std::ostream &os) {
  os << termcolor::reset << termcolor::bold << termcolor::red << "›"
     << termcolor::reset;
  return os;
};

} // namespace formatting
} // namespace shabam